crates/proof-of-sql/src/sql/logical_plans/expr.rs

```diff
@@ -0,0 +1,116 @@
+use super::LogicalPlanError;
+use crate::base::database::{ColumnRef, LiteralValue};
+use alloc::{boxed::Box, format};
+use serde::{Deserialize, Serialize};
+use sqlparser::ast::BinaryOperator as SqlBinaryOperator;
+
+/// Enum of column expressions that are either provable or supported in postprocessing
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum Expr {
+    /// Column
+    Column(ColumnRef),
+    /// Provable CONST expression
+    Literal(LiteralValue),
+    /// Binary operation
+    Binary {
+        /// Left hand side of the binary operation
+        left: Box<Expr>,
+        /// Right hand side of the binary operation
+        right: Box<Expr>,
+        /// Binary operator
+        op: BinaryOperator,
+    },
+    /// NOT expression
+    Not(Box<Expr>),
+}
+
+/// Enum of binary operators we support
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum BinaryOperator {
+    /// Equals
+    Eq,
+    /// Not equals
+    Neq,
+    /// Greater than
+    Gt,
+    /// Less than
+    Lt,
+    /// Greater than or equals
+    Gte,
+    /// Less than or equals
+    Lte,
+    /// Logical AND
+    And,
+    /// Logical OR
+    Or,
+    /// Add
+    Add,
+    /// Subtract
+    Sub,
+    /// Multiply
+    Mul,
+    /// Divide
+    Div,
+}
+
+impl TryFrom<SqlBinaryOperator> for BinaryOperator {
+    type Error = LogicalPlanError;
+
+    fn try_from(op: SqlBinaryOperator) -> Result<Self, Self::Error> {
+        match op {
+            SqlBinaryOperator::Eq => Ok(Self::Eq),
+            SqlBinaryOperator::NotEq => Ok(Self::Neq),
+            SqlBinaryOperator::Gt => Ok(Self::Gt),
+            SqlBinaryOperator::Lt => Ok(Self::Lt),
+            SqlBinaryOperator::GtEq => Ok(Self::Gte),
+            SqlBinaryOperator::LtEq => Ok(Self::Lte),
+            SqlBinaryOperator::And => Ok(Self::And),
+            SqlBinaryOperator::Or => Ok(Self::Or),
+            SqlBinaryOperator::Plus => Ok(Self::Add),
+            SqlBinaryOperator::Minus => Ok(Self::Sub),
+            SqlBinaryOperator::Multiply => Ok(Self::Mul),
+            SqlBinaryOperator::Divide => Ok(Self::Div),
```

Comment:
Can we make all of these names the same? Is there any reason to name things differently from `sqlparser`?

(For the record, I do think `Plus` and `Minus` are weird names, but the upsides of consistency outweigh that I think.)